File: dotnet/src/Core/Models/Entities/Host.cs
using Agience.Core.Models.Entities.Abstract;
using Agience.Core.Models.Enums;
using System.Text.Json.Serialization;

namespace Agience.Core.Models.Entities
{
    public class Host : PublicEntity
    {
        [JsonPropertyName("agents")]
        public virtual List<Agent> Agents { get; set; } = new();

        [JsonPropertyName("plugins")]
        public virtual List<Plugin> Plugins { get; set; } = new();

    }
}

File: python/src/core/models/entities/host.py
from pydantic import Field
from typing import List
from core.models.entities.abstract.public_entity import PublicEntity
from core.models.entities.agent import Agent
from core.models.entities.plugin import Plugin


class Host(PublicEntity):
    agents: List[Agent] = Field(default_factory=list, alias="agents")
    plugins: List[Plugin] = Field(default_factory=list, alias="plugins")

    class Config:
        allow_population_by_field_name = True
